### PR TITLE
Eapp 7 implement login page

### PR DESCRIPTION
**Pull request by LyubooShv**

### CODE REVIEW

**Comment by ivankraev** [comment]
https://user-images.githubusercontent.com/89323943/162904851-21c8815f-c50d-4fe4-a856-6e7ebdf970d1.mp4

File: src/App.tsx [hunk 1]
```diff
@@ -10,12 +10,13 @@ import "./App.global.css";
 
 const App = () => {
   const element = useRoutes(routes);
+  const location = useLocation();
   return (
     <ThemeProvider theme={appTheme}>
       <Box>
-        <Header />
+        {location.pathname !== "/login" && <Header />}
```
Comment (mariyaniliev) [inline src/App.tsx]: Create a variable for this part location.pathname !== "/login"

Example: 
const isLoggedIn =  location.pathname !== "/login"

 { isLoggedIn && <Header />}

File: src/App.tsx [hunk 2]
```diff
@@ -10,12 +10,13 @@ import "./App.global.css";
...
         <Box sx={{ display: "flex" }}>
-          <SideBar />
+          {location.pathname !== "/login" && <SideBar />}
```
Comment (mariyaniliev) [inline src/App.tsx]: The same as above one

File: src/pages/Login/CustomButton/CustomSubmitButton.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  ButtonProps,
+  CircularProgress,
+  CircularProgressProps,
+} from "@mui/material";
+
+export type SubmitButtonProps = {
+  loading?: boolean;
+  label?: string;
+  loadingColor?: CircularProgressProps["color"];
+} & ButtonProps;
```
Comment (mariyaniliev) [inline src/pages/Login/CustomButton/CustomSubmitButton.tsx]: Create a different file for "interfaces"

File: src/pages/Login/CustomInput/CustomInput.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+export const CustomInput = (props: InputUnstyledProps) => {
+  return (
+    <>
+      <InputUnstyled
+        style={{
+          width: "65%",
+          position: "relative",
+          display: "flex",
+          alignItems: "center",
+        }}
```
Comment (mariyaniliev) [inline src/pages/Login/CustomInput/CustomInput.tsx]: Put styles in the CSS file.

If you add just one style it is ok to be inline but if they are more than one, move them to the CSS file

File: src/pages/Login/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+import { useApiClient } from "../../utils/client";
+import { useNavigate } from "react-router-dom";
+import { AxiosResponse } from "axios";
+import { User } from "types/user";
+import logo from "../../theme/assets/gs-logo.png";
+import leftDraw from "../../theme/assets/draw_left.png";
+import rightDraw from "../../theme/assets/draw_right.png";
+import MailOutlineIcon from "@mui/icons-material/MailOutline";
+import LockOpenIcon from "@mui/icons-material/LockOpen";
+import CustomSubmitButton from "./CustomButton/CustomSubmitButton";
```
Comment (mariyaniliev) [inline src/pages/Login/Login.tsx]: Order imports, sort them by internal, external, CSS, redux

Ask Georgi or Lybo to help you with this one

File: src/pages/Login/Login.tsx [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+  });
+  const { isLoading } = useAppSelector((state: RootState) => state.user);
+  const { loginStart, loginSuccess, loginFail } = userActions();
+  const client = useApiClient();
+  const navigate = useNavigate();
+  const handleLogin = async (event: React.FormEvent<HTMLFormElement>) => {
+    const { email, password } = userCredentials;
+    event.preventDefault();
+    const error = validation(email, password);
+    if (Object.keys(error).length > 0) {
```
Comment (mariyaniliev) [inline src/pages/Login/Login.tsx:41]: Add a comment on this statement to be more clearly about what are you trying to achieve

File: src/pages/Login/Login.tsx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+        <Box sx={styles.formContainer}>
+          <Typography sx={styles.header}>Sign in to our platform</Typography>
+          <form
+            style={{
+              width: "100%",
+              display: "flex",
+              flexDirection: "column",
+              alignItems: "center",
+            }}
```
Comment (mariyaniliev) [inline src/pages/Login/Login.tsx:97]: Move the syles to CSS file

File: src/pages/Login/Login.tsx [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+              onFocus={handleFocus}
+              onBlur={handleBlur}
+              onChange={handleChange}
+              endAdornment={<MailOutlineIcon sx={styles.icons} />}
+            />
+
+            <Typography
+              sx={{
+                ...styles.helperText,
+                color: focused.password ? THEME_COLORS.purple : "",
```
Comment (mariyaniliev) [inline src/pages/Login/Login.tsx]: Create a variable for this one  - color: focused.password ? THEME_COLORS.purple : "",

Example:  const isPasswordFocusedStyle =  focused.password ? THEME_COLORS.purple : "",

color: isPasswordFocusedStyle

File: src/pages/Login/Login.tsx [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+            )}
+            <Box sx={styles.signInAttributes}>
+              <Box sx={styles.helpers}>
+                <Box sx={styles.checkboxHolder}>
+                  <Checkbox
+                    name="rememberMe"
+                    sx={styles.checkbox}
+                    onChange={(e) => {
+                      console.log(e.target.checked);
+                    }}
```
Comment (mariyaniliev) [inline src/pages/Login/Login.tsx:146]: Is this behavior expected?

File: src/pages/Login/Validation.ts [hunk 10]
```diff
@@ -0,0 +1,20 @@
+interface errorsValue {
+  emailError: string;
+  passwordError: string;
+}
```
Comment (mariyaniliev) [inline src/pages/Login/Validation.ts]: Move the interfaces to a different file

File: src/pages/Login/Validation.ts [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+export const validation = (email: string, password: string) => {
+  const regex = /^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i;
```
Comment (mariyaniliev) [inline src/pages/Login/Validation.ts]: Add a comment on what precisely this regex is doing or delete it and create it in a way that all of your teammates will understand

File: src/pages/Login/CustomInput/customInput-styles.ts [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import { styled } from "@mui/system";
+
+export const StyledInputElement = styled("input")`
+  width: 100%;
+  color: #979797;
```
Comment (Georgi07) [inline src/pages/Login/CustomInput/customInput-styles.ts]: This color is "grey03" in our design system

File: src/pages/Login/CustomInput/customInput-styles.ts [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+  box-shadow: 0px 3px 6px #00000029;
+  height: 38px;
+  border: none;
+  outline: none;
+  border-radius: 32px;
+  padding-left: 45px;
+  padding-right: 25px;
+`;
```
Comment (Georgi07) [inline src/pages/Login/CustomInput/customInput-styles.ts]: You can use styled service like: 
styled("input")({
width: "100%",
color: THEME_COLORS.grey03,
});

File: src/pages/Login/login-styles.ts [hunk 14]
```diff
@@ -0,0 +1,90 @@
+import { THEME_COLORS } from "../../theme/theme-constants";
+
+export const styles = {
+  mainContainer: {
+    display: "flex",
+    flexDirection: "row",
+    justifyContent: "stretch",
+    alignItems: "center",
+    height: "100vh",
+    backgroundColor: "#FAFAFA",
```
Comment (Georgi07) [inline src/pages/Login/login-styles.ts:10]: THEME_COLORS.white

File: src/pages/Login/login-styles.ts [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+  },
+
+  formContainer: {
+    display: "flex",
+    flexDirection: "column",
+    width: { xs: "70%", md: "100%" },
+    height: "50%",
+    borderRadius: "40px",
+    boxShadow: `0px 3px 6px ${THEME_COLORS.grey05}`,
+    backgroundColor: "#FFFFFF",
```
Comment (Georgi07) [inline src/pages/Login/login-styles.ts]: THEME_COLORS.white

File: src/pages/Login/Login.tsx [hunk 16]
```diff
@@ -0,0 +1,165 @@
...
+            <CustomInput
+              name="password"
+              type="password"
+              onChange={handleChange}
+              onBlur={handleBlur}
+              onFocus={handleFocus}
+              endAdornment={<LockOpenIcon sx={styles.icons} />}
+            />
+            {errorMessage && (
+              <Typography sx={{ mt: 1 }} color="#cc4949">
```
Comment (Georgi07) [inline src/pages/Login/Login.tsx]: Feel free to add this color in design system and we will use it in future

**Review by Georgi07 — CHANGES_REQUESTED** [review]

File: src/pages/Login/Login.tsx (hunk 8 above)
Comment (ivankraev) [reply src/pages/Login/Login.tsx]: example

File: src/pages/Login/Login.tsx (hunk 9 above)
Comment (ivankraev) [reply src/pages/Login/Login.tsx:146]: I will check how to implement "remember me" functionallity and them I will fix it.

**Pull request merged**